src/middleware/aws/instrumentation.rs

```diff
@@ -14,12 +14,28 @@ where
     }
 }
 
+/// Trait for instrumenting AWS futures with automatic span management.
+///
+/// This trait provides a convenient way to wrap AWS SDK futures with OpenTelemetry
+/// instrumentation, automatically handling span creation, error recording, and cleanup.
```

Comment:
Let's include here an example from "`AwsInstrumented` trait" section in the `README.md` to illustrate how `AwsInstrument` could be used to instrument AWS SDK operations.